apps/AppWithWearable/lib/pages/capture/page.dart

```diff
@@ -262,22 +271,109 @@ class CapturePageState extends State<CapturePage> with AutomaticKeepAliveClientM
   @override
   void initState() {
     btDevice = widget.device;
+    WidgetsBinding.instance.addObserver(this);
     SchedulerBinding.instance.addPostFrameCallback((_) async {
       debugPrint('SchedulerBinding.instance');
       initiateBytesProcessing();
+      if (await LocationService().isServiceEnabled() == false ||
+          await LocationService().permissionStatus() != PermissionStatus.granted) {
+        showDialog(
+          context: context,
+          builder: (c) => getDialog(
+            context,
+            () {
+              setState(() {
+                shouldAskPermissionAgain = false;
+              });
+              Navigator.of(context).pop();
+            },
+            () async {
+              Navigator.of(context).pop();
+              setState(() {
+                shouldAskPermissionAgain = true;
+              });
+              await requestLocationPermission();
+            },
+            'Know where your memories were created! 🌍',
+            '\nWe can use your location to add a location tag to your memories. This will help you remember where you were when you created them.\n\nWould you like to enable location services?',
+            singleButton: false,
+          ),
+        );
+      }
     });
     _processCachedTranscript();
     // processTranscriptContent(context, '''a''', null);
     super.initState();
   }
 
+  @override
+  void didChangeAppLifecycleState(AppLifecycleState state) async {
```

Comment:
Do we need to do this check every time the app resumes?

Reply:
It totally depends on us. I can remove it if suggested.

I've added that check to detect if the user has changed location settings (in iOS) after minimising the app and then came back to the app. If the user denies permission, then this won't get called because the condition will never be true.